tasks/prepareSiteFiles.js

```diff
@@ -0,0 +1,257 @@
+  
+      pokemonNameMap[id] = getFormNameOfProblematicPokemon(id);
+      return pokemonNameMap;
+    } catch (e) {
+      throw Error(`${currentPokemon.id} - ${e}`);
+    }
+  }
+
+  function getFormNameOfProblematicPokemon(id = 0) {
+    switch (id) {
+      case 1242:
+        return 'Ash-Greninja';
+      case 1285:
+        return 'Meowstic-F';
+      case 1310:
+        return 'Rockruff Own-Tempo';
+      case 1441:
+        return 'Indeedee-F';
+      case 1454:
+        return 'Basculegion-F';
+      case 1456:
+        return 'Oinkologne-F';
+      default:
+        throw Error(`Bad Pokemon ID in PokemonNameMap: ${id}`);
+    }
+  }
+
+function basicFileHandler(currentItem) {
+    if(currentItem.labelName === "") return createDataObject("", "");
+    return createDataObject(currentItem.labelName, currentItem.wordDataArray[0].str);
+}
+
+function createDataObject(labelName, data) {
+    return [labelName, data];
+}
+
+function minifyItemObject(itemData) {
+    const {no, type, iconid, price, bp_price, group_id, flags0} = itemData;
+    return [
+        no, type, iconid, price, bp_price, group_id, flags0
+    ];
+}
+
+function minifyPersonalObject(p) {
+    return Object.values(p);
+}
+
+function minifyTrainerTypeObject(t) {
+    return {
+        ID: t.TrainerID,
+        Class: t.LabelTrType
+    }
+}
+
+function minifyTrainerDataObject(t) {
+    return {
+        TypeID: t.TypeID,
+        Name: t.NameLabel,
+        money: t.Gold
+    }
+}
+
+function minifyTrainerPartyObject(t) {
+    let party = [];
+   
+    for(let i = 1; i < 7; i++) {
+        if(t[`P${i}MonsNo`] === 0) continue;
+        const mon = [
+            t[`P${i}MonsNo`], t[`P${i}FormNo`], t[`P${i}Level`], t[`P${i}Sex`], t[`P${i}Seikaku`],
+            t[`P${i}Tokusei`], t[`P${i}Waza1`], t[`P${i}Waza2`], t[`P${i}Waza3`], t[`P${i}Waza4`],
+            t[`P${i}Item`], 
+            t[`P${i}TalentHp`], t[`P${i}TalentAtk`], t[`P${i}TalentDef`],
+            t[`P${i}TalentSpAtk`], t[`P${i}TalentSpDef`], t[`P${i}TalentAgi`], 
+            t[`P${i}EffortHp`], t[`P${i}EffortAtk`], t[`P${i}EffortDef`],
+            t[`P${i}EffortSpAtk`], t[`P${i}EffortSpDef`], t[`P${i}EffortAgi`]
+        ];
+
+        party.push(mon);
+    }
+
+    return party;
+}
+
+function minifyTrainerTable() {
+    const t = {};
+    t.TrainerType = TrainerTable.TrainerType.map(minifyTrainerTypeObject);
+    t.TrainerData = TrainerTable.TrainerData.map(minifyTrainerDataObject).slice(0, 2000);
+    t.TrainerPoke = TrainerTable.TrainerPoke.map(minifyTrainerPartyObject).slice(0, 2000);
+    return t
+}
+
+function minifyItemTable() {
+    const t = {};
+    t.Item = ItemTable.Item.map(minifyItemObject);
+    t.WazaMachine = t.WazaMachine
+    return t;
+}
+
+function minifyPersonalTable() {
+    return PersonalTable.Personal.map(minifyPersonalObject);
+}
+
+function writeToOutput(filename, data) {
+    fs.writeFileSync(path.join(__dirname, '..', 'output', filename), JSON.stringify(data), 'utf-8');
```

Comment:
You could use the space parameter for the stringify here (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#using_the_space_parameter). Should make it a bit easier to read the resulting json files from this. It also might add some length which might not be preferable.

Reply:
Yeah the goal here is absolute minification. you can also use numbers to specify how many spaces (or was it tabs?) you want per indentation. Our old content used 2